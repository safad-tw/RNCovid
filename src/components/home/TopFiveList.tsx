import * as React from 'react';
import {
    Text,
    View,
    StyleSheet,
    Dimensions, FlatList, Button
} from 'react-native';

import LanguagesStrings from './../../localization/LanguagesStrings';
import {Country} from './../../models/Country'

interface TopFiveListProps {
    data: [Country],
    navigateTo: () => void;
}

class TopFiveList extends React.Component<TopFiveListProps> {

    constructor(props: TopFiveListProps) {
        super(props);
    }

    _renderItem(data: any) {
        return (
            <Text style={styles.cardListName}>
                {data.index + 1}. {data.item.Country}
            </Text>
        )
    }

    goToViewAllScreen() {
        this.props.navigateTo()
    }

    render() {
        return (
            <View style={styles.cardView}>
                <FlatList
                    data={this.props.data}
                    renderItem={this._renderItem}
                    keyExtractor={(item) => item.CountryCode}
                />
                <Button
                    title={LanguagesStrings.viewAll}
                    color="blue"
                    onPress={() => this.goToViewAllScreen()}
                />
            </View>
        )
    }
}

const width = Dimensions.get('window').width - 20;

const styles = StyleSheet.create({
    cardView: {
        height: 200,
        width: width,
        margin: 5,
        justifyContent: 'center',
        backgroundColor: '#F5FCFF',
        borderRadius: 5,
        shadowColor: 'gray',
        shadowOffset: { width: 0, height: 1 },
        shadowOpacity: 0.8,
        shadowRadius: 2,
        elevation: 5
    },
    cardListName: {
        margin: 5,
        textAlign: 'left',
        fontSize: 17,
        fontFamily: "Avenir-medium"
    }

})

export default TopFiveList;