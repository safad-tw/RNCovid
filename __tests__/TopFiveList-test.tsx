import * as React from "react";
import renderer from "react-test-renderer";
import TopFiveList from "../src/components/home/TopFiveList";
import { render, fireEvent } from '@testing-library/react-native';
import '@testing-library/jest-dom'
const data: any = [
  {
    ID: "d81abd15-3c81-4c98-82e7-6042959daa6d",
    Country: "Afghanistan",
    CountryCode: "AF",
    Slug: "afghanistan",
    NewConfirmed: 0,
    TotalConfirmed: 155093,
    NewDeaths: 0,
    TotalDeaths: 7201,
    NewRecovered: 0,
    TotalRecovered: 0,
    Date: "2021-09-29T12:04:59.613Z",
    Premium: {}
  },
  {
    ID: "39714019-ce94-461b-b81a-1236f961c02a",
    Country: "Albania",
    CountryCode: "AL",
    Slug: "albania",
    NewConfirmed: 0,
    TotalConfirmed: 168782,
    NewDeaths: 0,
    TotalDeaths: 2668,
    NewRecovered: 0,
    TotalRecovered: 0,
    Date: "2021-09-29T12:04:59.613Z",
    Premium: {}
  },
  {
    ID: "729b2b5d-848c-4597-98b2-d35a90045c80",
    Country: "Algeria",
    CountryCode: "DZ",
    Slug: "algeria",
    NewConfirmed: 0,
    TotalConfirmed: 203045,
    NewDeaths: 0,
    TotalDeaths: 5797,
    NewRecovered: 0,
    TotalRecovered: 0,
    Date: "2021-09-29T12:04:59.613Z",
    Premium: {}
  },
  {
    ID: "da4c9b7a-5f18-4105-a349-0af456ebc00b",
    Country: "Andorra",
    CountryCode: "AD",
    Slug: "andorra",
    NewConfirmed: 0,
    TotalConfirmed: 15192,
    NewDeaths: 0,
    TotalDeaths: 130,
    NewRecovered: 0,
    TotalRecovered: 0,
    Date: "2021-09-29T12:04:59.613Z",
    Premium: {}
  },
  {
    ID: "e0737d69-f4ee-417d-b781-83ceb157efd9",
    Country: "Angola",
    CountryCode: "AO",
    Slug: "angola",
    NewConfirmed: 0,
    TotalConfirmed: 56040,
    NewDeaths: 0,
    TotalDeaths: 1526,
    NewRecovered: 0,
    TotalRecovered: 0,
    Date: "2021-09-29T12:04:59.613Z",
    Premium: {}
  }];
test("Render component", () => {
  const clickFn = jest.fn();
  const tree = renderer.create(<TopFiveList data={data} navigateTo={clickFn} />).toJSON();
  expect(tree).not.toBeNull()
});

test("Render view all button", () => {

  const clickFn = jest.fn();
  const tree = renderer.create(<TopFiveList data={data} navigateTo={clickFn} />).toJSON();
  expect(tree).not.toBeNull()
});

test('Renders snapshot as expected', () => {
  const clickFn = jest.fn();
  const tree = renderer.create(<TopFiveList data={data} navigateTo={clickFn} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Click view all', () => {

  const clickFn = jest.fn();
  const { getByText } = render(<TopFiveList data={data} navigateTo={clickFn} />)
  fireEvent.press(getByText("View All"));

});

test('Click view all and verify call back is called', () => {

  const mockcallBack = jest.fn();
  const { getByText } = render(<TopFiveList data={data} navigateTo={mockcallBack} />)
  fireEvent.press(getByText("View All"));
  expect(mockcallBack).toHaveBeenCalled();


});