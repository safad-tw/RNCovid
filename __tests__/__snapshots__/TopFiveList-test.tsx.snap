// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "borderRadius": 5,
      "elevation": 5,
      "height": 200,
      "justifyContent": "center",
      "margin": 5,
      "shadowColor": "gray",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 2,
      "width": 730,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "Country": "Afghanistan",
          "CountryCode": "AF",
          "Date": "2021-09-29T12:04:59.613Z",
          "ID": "d81abd15-3c81-4c98-82e7-6042959daa6d",
          "NewConfirmed": 0,
          "NewDeaths": 0,
          "NewRecovered": 0,
          "Premium": Object {},
          "Slug": "afghanistan",
          "TotalConfirmed": 155093,
          "TotalDeaths": 7201,
          "TotalRecovered": 0,
        },
        Object {
          "Country": "Albania",
          "CountryCode": "AL",
          "Date": "2021-09-29T12:04:59.613Z",
          "ID": "39714019-ce94-461b-b81a-1236f961c02a",
          "NewConfirmed": 0,
          "NewDeaths": 0,
          "NewRecovered": 0,
          "Premium": Object {},
          "Slug": "albania",
          "TotalConfirmed": 168782,
          "TotalDeaths": 2668,
          "TotalRecovered": 0,
        },
        Object {
          "Country": "Algeria",
          "CountryCode": "DZ",
          "Date": "2021-09-29T12:04:59.613Z",
          "ID": "729b2b5d-848c-4597-98b2-d35a90045c80",
          "NewConfirmed": 0,
          "NewDeaths": 0,
          "NewRecovered": 0,
          "Premium": Object {},
          "Slug": "algeria",
          "TotalConfirmed": 203045,
          "TotalDeaths": 5797,
          "TotalRecovered": 0,
        },
        Object {
          "Country": "Andorra",
          "CountryCode": "AD",
          "Date": "2021-09-29T12:04:59.613Z",
          "ID": "da4c9b7a-5f18-4105-a349-0af456ebc00b",
          "NewConfirmed": 0,
          "NewDeaths": 0,
          "NewRecovered": 0,
          "Premium": Object {},
          "Slug": "andorra",
          "TotalConfirmed": 15192,
          "TotalDeaths": 130,
          "TotalRecovered": 0,
        },
        Object {
          "Country": "Angola",
          "CountryCode": "AO",
          "Date": "2021-09-29T12:04:59.613Z",
          "ID": "e0737d69-f4ee-417d-b781-83ceb157efd9",
          "NewConfirmed": 0,
          "NewDeaths": 0,
          "NewRecovered": 0,
          "Premium": Object {},
          "Slug": "angola",
          "TotalConfirmed": 56040,
          "TotalDeaths": 1526,
          "TotalRecovered": 0,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontFamily": "Avenir-medium",
              "fontSize": 17,
              "margin": 5,
              "textAlign": "left",
            }
          }
        >
          1
          . 
          Afghanistan
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontFamily": "Avenir-medium",
              "fontSize": 17,
              "margin": 5,
              "textAlign": "left",
            }
          }
        >
          2
          . 
          Albania
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontFamily": "Avenir-medium",
              "fontSize": 17,
              "margin": 5,
              "textAlign": "left",
            }
          }
        >
          3
          . 
          Algeria
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontFamily": "Avenir-medium",
              "fontSize": 17,
              "margin": 5,
              "textAlign": "left",
            }
          }
        >
          4
          . 
          Andorra
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontFamily": "Avenir-medium",
              "fontSize": 17,
              "margin": 5,
              "textAlign": "left",
            }
          }
        >
          5
          . 
          Angola
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "blue",
            },
          ]
        }
      >
        View All
      </Text>
    </View>
  </View>
</View>
`;
